src/EFCore.PG/Storage/Internal/NpgsqlRelationalConnection.cs

```diff
@@ -35,6 +35,8 @@ protected override DbConnection CreateDbConnection()
                 conn.ProvideClientCertificatesCallback = ProvideClientCertificatesCallback;
             if (RemoteCertificateValidationCallback != null)
                 conn.UserCertificateValidationCallback = RemoteCertificateValidationCallback;
+            if (conn.Settings.MaxAutoPrepare == 0)
+                Dependencies.ConnectionLogger.AutoPrepareDisabledWarning(conn, ConnectionId);
```

Comment:
I definitely don't think we should log every single time a connection is opened - one warning at startup should really be enough, otherwise we're really polluting the logs.